### PR TITLE
XCL-2663 Unify makefile extensions

### PR DESCRIPTION
**Pull request by milgra**
[DESCRIPTION]
- added *.mk extensions
- changed extensions in markdown documentation
- modified makefiles to stop in case of invalid preset to avoid accidental misbuilds
- makefile help fixes

[ JIRA ]
https://jira.3amlabs.net/browse/XCL-2663

[REVIEWERS]

[QA]
All preset combinations build
tests build
Travis tests pass

[RELEASE NOTES]
Makefile extensions unified. Make command stops in case of invalid preset with description.

### CODE REVIEW

**Comment by atigyi** [comment]
Approved, awesometown!!

File: Makefile [hunk 1]
```diff
@@ -24,15 +24,15 @@ XI_BSP_TLS ?= wolfssl
...
 include make/mt-config/mt-presets.mk
 
-include make/mt-config/mt-config
-include make/mt-os/mt-os
-include make/mt-os/mt-$(XI_CONST_PLATFORM_CURRENT)
-include make/mt-config/mt-examples
-include make/mt-config/tests/mt-tests-tools
-include make/mt-config/tests/mt-tests-unit
-include make/mt-config/tests/mt-tests-integration
+include make/mt-config/mt-config.mk
```
Comment (olgierdh) [inline Makefile]: This is awesome! Finally my editors will gimme rainbows and unicorns!

File: make/mt-config/mt-help.mk [hunk 2]
```diff
@@ -67,17 +67,20 @@ help_build_presets:
...
-	$(info .   CC3200_REL_MIN_UNSECURE)
-	$(info .   CC3200_REL)
-	$(info . )
+	$(info .   ARM_REL_MIN)
+	$(info .   ARM_REL)
+	$(info .   CC3200)
+	$(info .   CC3200_SDK120)
+	$(info .   CC3200_TLS_SOCKET)
+	$(info .   STM32F4)
+	$(info .   FUZZ_TESTS)
```
Comment (olgierdh) [inline make/mt-config/mt-help.mk]: I'm not sure if we should reveal FUZZ_TESTS for now... This is premature as this only works in certain conditions etc. I don't know. @DellaBitta what do you think ?

File: make/mt-config/mt-presets.mk [hunk 3]
```diff
@@ -119,10 +119,15 @@ $(error Fuzz testing won\'t work on OSX)
...
-TARGET ?= $(TARGET_STATIC_REL)
-  $(info INFO: PRESET: '$(PRESET)' not recognised, using default CONFIG: [$(CONFIG)] and TARGET: [$(TARGET)])
+    ifndef PRESET
+    # default settings in case of undefined
+    CONFIG ?= $(CONFIG_POSIX_MIN)
+    TARGET ?= $(TARGET_STATIC_REL)
+  	    $(info INFO: '$(PRESET)' not detected, using default CONFIG: [$(CONFIG)] and TARGET: [$(TARGET)])
+    else
+    # error in case of unrecognised PRESET
+    $(error Invalid PRESET, see valid presets in make/mt-config/mt-presets.mk)
```
Comment (olgierdh) [inline make/mt-config/mt-presets.mk]: Cool! It would be nice to show the list of available presets here as well.

Comment (milgra) [inline make/mt-config/mt-presets.mk]: That was my first idea but Apri says that we should keep it simple. In this case we don't have to maintain this part with every new preset :)

Comment (olgierdh) [inline make/mt-config/mt-presets.mk]: Ok. No further questions ;)

**Pull request merged**